src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -122,19 +123,61 @@ public static bool CanRemoveParentheses(this ParenthesizedExpressionSyntax node)
                 : false;
         }
 
+        [ThreadStatic]
+        private static Stack<SyntaxNode> s_nodeStack;
```

Comment:
This is potentially called for every single parenthesized expression. Seems a bad idea to allocate a stack each time.